service/controller/chart/v1/resource/release/resource.go

```diff
@@ -89,6 +93,39 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func (r *Resource) updateAnnotations(ctx context.Context, cr v1alpha1.Chart, releaseState ReleaseState) error {
```

Comment:
This function do 3 different things. 
1. Get the current Chart. 
2. Compare md5sum
3. Update `Chart` with a new one. 
But just name it `updateAnnotation` is a bit unclear for me. 
Can we divide the function of inside, or simply rename it?

Reply:
The comparison is so we only make a k8s API call if the annotation has changed. The reconciliation logic uses `isReleaseModified`. Should I add more comments to make that clearer?

Reply:
Maybe some `godoc` for the function is enough then.